pkg/jobs/metricspoller/job_statistics.go

```diff
@@ -124,13 +130,14 @@ func manageProtectedTimestamps(ctx context.Context, execCtx sql.JobExecContext)
 				if err := processSchedulePTSRecord(ctx, jobspb.ScheduleID(id), rec, schedulePtsStats, txn); err != nil {
 					return err
 				}
-			default:
-				continue
 			}
+			return nil
+		}); err != nil {
+			// If we fail to process one record, we should still try to process
+			// subsequent records, therefore, just log the error instead of returning
+			// early.
+			log.Infof(ctx, "Could not process pts record id %d: %s", rec.ID, err.Error())
```

Comment:
[nit] we usually don't capitalize start log messages unless it is a proper noun or abbreviation

Reply:
you remind me of this every time you review my code :D